src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -937,6 +932,61 @@ type internal TypeCheckInfo
 
     let DefaultCompletionItem item = CompletionItem ValueNone ValueNone item
 
+    let CompletionItemSuggestedName displayName =
+        {
+            ItemWithInst = ItemWithNoInst(Item.NewDef(Ident(displayName, range0)))
+            MinorPriority = 0
+            Type = None
+            Kind = CompletionItemKind.SuggestedName
+            IsOwnMember = false
+            Unresolved = None
+        }
+
+    /// Check whether the suggested name is unused and add it to the list.
+    /// In the future we could use an increasing numeric suffix for conflict resolution
+    let PostProcessSuggestedPatternName (pos: pos) name list =
+        let name = String.lowerCaseFirstChar name
+
+        let unused =
+            sResolutions.CapturedNameResolutions
+            |> ResizeArray.forall (fun r ->
+                match r.Item with
+                | Item.Value vref when r.Pos.Line = pos.Line -> vref.DisplayName <> name
+                | _ -> true)
+
+        if unused then
+            CompletionItemSuggestedName name :: list
+        else
+            list
+
+    /// Suggest name based on type, add it to the list
+    let SuggestNameBasedOnType g pos ty list =
+        if isNumericType g ty then
+            PostProcessSuggestedPatternName pos "num" list
+        else
+            match tryTcrefOfAppTy g ty with
+            | ValueSome tcref -> PostProcessSuggestedPatternName pos tcref.DisplayName list
+            | _ -> list
```

Comment:
Suggesting `num` for numbers. Let me know what you think.

Reply:
I think it's good